File: src/main/java/com/tibagni/logviewer/updates/ReleaseInfo.java
package com.tibagni.logviewer.updates;

import org.json.JSONObject;

public class ReleaseInfo {
  private String versionName;
  private double version;
  private String releaseUrl;
  private String releaseNotes;

  ReleaseInfo(JSONObject json) throws InvalidReleaseException {
    versionName = json.getString("tag_name");
    releaseUrl = json.getString("html_url");
    releaseNotes = json.getString("body");

    try {
      version = Double.parseDouble(versionName);
    } catch (NumberFormatException nfe) {
      throw new InvalidReleaseException("Invalid Release " + versionName);
    }
  }

  public String getVersionName() {
    return versionName;
  }

  public double getVersion() {
    return version;
  }

  public String getReleaseUrl() {
    return releaseUrl;
  }

  public String getReleaseNotes() {
    return releaseNotes;
  }
}

File: src/main/java/com/tibagni/logviewer/updates/ReleaseInfo.kt
package com.tibagni.logviewer.updates

import org.json.JSONObject

class ReleaseInfo internal constructor(json: JSONObject) {
  val versionName: String = json.getString("tag_name")
  val releaseUrl: String = json.getString("html_url")
  val releaseNotes: String = json.getString("body")
  val version: Double = try {
    versionName.toDouble()
  } catch (nfe: NumberFormatException) {
    throw InvalidReleaseException("Invalid Release $versionName")
  }
}